### PR TITLE
Feature/release workflow

### PR DESCRIPTION
**Pull request by a14m**
## Proposed changes
Add CircleCI integration to enable testing and publishing of new versions on npm
The change includes some minor changes to the `package.json` scripts (to follow the same conventions that we are using for other projects)
These conventions include using `_script` for scripts that are supposed to be used internally by `yarn` and won't be usually called by the user...

rename the `lint` and `test` yarn tasks to be clearer
add `lint:fix` to try to automate fixing linting/prettier issues automatically
add `test:ci` that is specific to run on CircleCI since the normal `yarn test` will trigger watch (which won't be working on CI services)

Updated the documentation to include notes about the publishing workflow, and usage (using both npm or github) to add the dependency to other projects

## Ticket
- https://app.asana.com/0/1154394619859001/1164622600639222/f

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,18 +1,20 @@
 {
   "name": "sepera-storybook",
-  "version": "0.1.0",
+  "version": "0.0.1-development",
   "private": true,
-  "dependencies": {
-    "react": "^16.13.0",
-    "react-dom": "^16.13.0"
+  "engines": {
```
Comment (cawo88) [inline package.json:5]: 👍

**Comment by cawo88** [comment]
Thanks for adding CircleCI. I had some missing devDependencies so I added them and will merge this PR.

File: package.json [hunk 2]
```diff
@@ -1,18 +1,20 @@
 {
-  "name": "sapera-storybook",
-  "version": "0.1.0",
+  "name": "sepera-storybook",
```
Comment (einSelbst) [inline package.json:2]: why back to sepera ? :)

Comment (a14m) [reply package.json:2]: it was done by mistake...
the original init commit added this by mistake https://github.com/infographicsgroup/sapera-storybook/blame/master/package.json#L2
but it was lost along the way (not rebasing on master after https://github.com/infographicsgroup/sapera-storybook/commit/49a6bf4df75746938c742c2eb18e81754286e583)
I'll fix it on master now

**Pull request merged**